src/main/webapp/app/course/course-scores/scores-storage.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { ScoresPerExerciseType } from 'app/entities/exercise.model';
+import { Result } from 'app/entities/result.model';
+import { CourseScores } from 'app/course/course-scores/course-scores';
+
+@Injectable({ providedIn: 'root' })
+export class ScoresStorageService {
+    private storedTotalScores: Map<number, CourseScores> = new Map();
+
+    private storedScoresPerExerciseType: Map<number, ScoresPerExerciseType> = new Map();
+
+    private participationResults: Result[] = [];
+
+    getStoredTotalScores(courseId: number): CourseScores | undefined {
+        return this.storedTotalScores.get(courseId);
+    }
+
+    setStoredTotalScores(courseId: number, totalScores: CourseScores): void {
+        this.storedTotalScores.set(courseId, totalScores);
+    }
+
+    getStoredScoresPerExerciseType(courseId: number): ScoresPerExerciseType | undefined {
+        return this.storedScoresPerExerciseType.get(courseId);
+    }
+
+    setStoredScoresPerExerciseType(courseId: number, scoresPerExerciseType: ScoresPerExerciseType): void {
+        this.storedScoresPerExerciseType.set(courseId, scoresPerExerciseType);
+    }
+
+    getStoredParticipationResult(participationId: number): Result | undefined {
+        return this.participationResults.filter((result) => result.participation?.id === participationId)[0];
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find might be a more elegant solution